main.cpp

```diff
@@ -32,8 +32,82 @@
  Wait for my code review.
  */
 
+// FloatType object definition
+struct FloatType
+{
+    float add(float lhs, float rhs ){ return lhs + rhs;}
+    float subtract(float lhs,float rhs ){ return lhs - rhs;}
+    float multiply(float lhs, float rhs ){ return lhs * rhs;}
+    float divide(float lhs, float rhs ){ return lhs / rhs;}
+}; 
+
+// DoubleType object definition
+struct DoubleType
+{
+    double add(double lhs, double rhs ){ return lhs + rhs;}
+    double subtract(double lhs,double rhs ){ return lhs - rhs;}
+    double multiply(double lhs, double rhs ){ return lhs * rhs;}
+    double divide(double lhs, double rhs ){ return lhs / rhs;}
+}; 
+
+// IntType object definition
+struct IntType
+{
+    int add(int lhs, int rhs ){ return lhs + rhs;}
+    int subtract(int lhs,int rhs ){ return lhs - rhs;}
+    int multiply(int lhs, int rhs ){ return lhs * rhs;}
+    int divide(int lhs, int rhs ){ return lhs / rhs;}
```

Comment:
You need to add a handler `if rhs == 0`.   integer division-by-zero will crash the program. 

- [ ] add `if(rhs == 0)` handler